### ISSUE TITLE
Data race in csi_attacher

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#19358]
```
WARNING: DATA RACE
Write at 0x00c42001c420 by goroutine 23:
  runtime.closechan()
      /usr/local/go/src/runtime/chan.go:313 +0x0
  github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch.(*FakeWatcher).Stop()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch/watch.go:110 +0x118
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.(*csiAttacher).waitForVolumeAttachmentInternal()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_obj_test/csi_attacher.go:231 +0x20b0
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.(*csiAttacher).waitForVolumeAttachment()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_obj_test/csi_attacher.go:135 +0x2b2
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:259 +0xcf1
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x00c42001c420 by goroutine 25:
  runtime.chansend()
      /usr/local/go/src/runtime/chan.go:128 +0x0
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment.func1()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch/watch.go:140 +0x13a

Goroutine 23 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  main.main()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_testmain.go:142 +0x2f0

Goroutine 25 (running) created at:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:251 +0x1185
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
==================
WARNING: DATA RACE
Write at 0x00c4201f6330 by goroutine 23:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:234 +0x4b6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x00c4201f6331 by goroutine 25:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment.func1()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:253 +0x79

Goroutine 23 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  main.main()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_testmain.go:142 +0x2f0

Goroutine 25 (running) created at:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:251 +0x1185
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
==================
WARNING: DATA RACE
Write at 0x00c4201f6338 by goroutine 23:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:234 +0x4b6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x00c4201f6338 by goroutine 25:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment.func1()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:252 +0x40

Goroutine 23 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  main.main()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_testmain.go:142 +0x2f0

Goroutine 25 (running) created at:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:251 +0x1185
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
==================
WARNING: DATA RACE
Write at 0x00c4201f6348 by goroutine 23:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:234 +0x4b6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x00c4201f6348 by goroutine 25:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment.func1()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:254 +0xb7

Goroutine 23 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  main.main()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_testmain.go:142 +0x2f0

Goroutine 25 (running) created at:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:251 +0x1185
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
==================
WARNING: DATA RACE
Write at 0x00c420092fa0 by goroutine 23:
  github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch.(*FakeWatcher).Reset()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch/watch.go:126 +0xc0
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:235 +0x507
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x00c420092fa0 by goroutine 25:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment.func1()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch/watch.go:140 +0x11f

Goroutine 23 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  main.main()
      github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/_test/_testmain.go:142 +0x2f0

Goroutine 25 (running) created at:
  github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment()
      /go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:251 +0x1185
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
panic: send on closed channel

goroutine 51 [running]:
github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch.(*FakeWatcher).Modify(...)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/apimachinery/pkg/watch/watch.go:140
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment.func1(0xc4201f6320, 0xc420294000, 0xc420092fa0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:255 +0x13b
created by github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi.TestAttacherWaitForVolumeAttachment
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi/csi_attacher_test.go:251 +0x1186
FAIL	github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/volume/csi	3.162s
```

@openshift/storage @liggitt post rebase

**Comment by liggitt** [comment]
Upstream issue?

**Comment by jsafrane** [comment]
I reproduced it upstream and filled https://github.com/kubernetes/kubernetes/issues/62630

**Comment by smarterclayton** [comment]
Looks like it, i don't see any way we can change it.

**Comment by jsafrane** [comment]
Fixed by #19508

/close
